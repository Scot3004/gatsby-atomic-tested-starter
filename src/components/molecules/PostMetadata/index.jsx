import React from 'react';
import PropTypes from 'prop-types';
import { IconContext } from 'react-icons';
import { FaCalendarAlt, FaUserClock } from 'react-icons/fa';
import MetadataElement from '../../atoms/MetadataElement';
import MetadataText from '../../atoms/MetadataText';
import MetadataContainer from '../../atoms/MetadataContainer';

export const PostMetadata = ({ date, timeToRead }) => (
  <IconContext.Provider value={{ style: { color: '#663399' } }}>
    <MetadataContainer>
      <MetadataElement>
        <FaCalendarAlt />
        <MetadataText>{date}</MetadataText>
      </MetadataElement>
      <MetadataElement>
        <FaUserClock />
        <MetadataText>{timeToRead} minutes to read</MetadataText>
      </MetadataElement>
    </MetadataContainer>
  </IconContext.Provider>
);

PostMetadata.propTypes = {
  date: PropTypes.string.isRequired,
  timeToRead: PropTypes.string.isRequired
};

export default PostMetadata;
