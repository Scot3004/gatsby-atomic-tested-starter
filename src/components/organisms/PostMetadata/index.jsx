import React from 'react';
import PropTypes from 'prop-types';
import MetadataElement from '../../atoms/MetadataElement';
import MetadataText from '../../atoms/MetadataText';
import MetadataContainer from '../../atoms/MetadataContainer';
import FeatherIcon from '../../molecules/FeatherIcon';

export const PostMetadata = ({ date, timeToRead }) => {
  return (
    <MetadataContainer>
      <MetadataElement>
        <FeatherIcon name="calendar" size={16} />
        <MetadataText>{date}</MetadataText>
      </MetadataElement>
      <MetadataElement>
        <FeatherIcon name="clock" size={16} />
        <MetadataText>{timeToRead} minutes to read</MetadataText>
      </MetadataElement>
    </MetadataContainer>
  );
};

PostMetadata.propTypes = {
  date: PropTypes.string.isRequired,
  timeToRead: PropTypes.string.isRequired
};

export default PostMetadata;
