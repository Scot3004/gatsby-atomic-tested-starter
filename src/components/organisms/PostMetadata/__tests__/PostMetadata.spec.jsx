import React from 'react';
import 'jest-styled-components';
import ShallowRenderer from 'react-test-renderer/shallow';
import PostMetadata from '..';

describe('When I want to see the metadata of a post', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<PostMetadata date="2019-06-22" timeToRead="3" />);
  test('Then the metadata should be displayed', () => {
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
