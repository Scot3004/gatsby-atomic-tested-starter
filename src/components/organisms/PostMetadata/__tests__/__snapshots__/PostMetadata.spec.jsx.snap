// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I want to see the metadata of a post Then the metadata should be displayed 1`] = `
<ForwardRef>
  <ForwardRef>
    <FeatherIcon
      name="calendar"
      size={16}
    />
    <ForwardRef>
      2019-06-22
    </ForwardRef>
  </ForwardRef>
  <ForwardRef>
    <FeatherIcon
      name="clock"
      size={16}
    />
    <ForwardRef>
      3
       minutes to read
    </ForwardRef>
  </ForwardRef>
</ForwardRef>
`;
